### ISSUE TITLE
TypeError: Headers constructor

### ISSUE THREAD

**Issue by Stiropor** [trandaison/nuxt-3-auth#30]
I've updated from 0.1.1 to latest ... and all stopped working. I'm using only this:
```ts
<Authenticator :credentials="{ email: '', password: '' }" />
```

All I'm getting is `An error has occurred`, no request to server or anything.

and this is my setup
```ts
auth: {
		endpoints: {
			baseUrl: process.env.PLATFORM_API_BASE_URL,
			login: { url: '/account/login', method: 'POST', property: '' },
			logout: { url: '/account/logout', method: 'DELETE' },
			refresh: { url: '/account/refresh', method: 'POST', property: '' },
			user: { url: '/account/me', method: 'GET', property: '' }
		},
		routes: {
			login: {
				name: 'login',
				file: resolve(__dirname, './pages/login.vue'),
				path: '/login'
			}
		},
		token: {
			headerName: 'Authorization',
			type: 'Bearer',
			property: 'accessToken',
			maxAge: parseInt(process.env.REFRESH_TOKEN_EXPIRY || '3600')
		},
		refreshToken: {
			paramName: 'refreshToken',
			property: 'refreshToken',
			maxAge: parseInt(process.env.REFRESH_TOKEN_EXPIRY || '3600')
		},
		middleware: {
			global: true
		},
		debug: true,
		useI18n: false,
		useGlobalFetch: false
	}
}
```

This exact setup used to work for 2 months now. 

I see some things about headers were added, but I don't need any extra headers.
What am I missing, am I the only one having this issue? Using Nuxt 3.12.1

**Comment by datafieber** [comment]
Have same issue!
Upgraded also from 0.1.1 to latest, with exact same NUXT version 3.12.4
After the upgrade the login function does not work anymore, i got also the error about the headers, concret:
```

Object { dep: {…}, __v_isRef: true, __v_isShallow: false, _rawValue: TypeError, _value: TypeError }
​
__v_isRef: true
​
__v_isShallow: false
​
_rawValue: TypeError: Headers constructor: Argument 1 could not be converted to any of: sequence<sequence<ByteString>>, record<ByteString, ByteString>.
​​
columnNumber: 12
​​
fileName: "http://localhost:3334/_nuxt/node_modules/ofetch/dist/shared/ofetch.4747642d.mjs?v=d0a42750"
​​
lineNumber: 117
​​
message: "Headers constructor: Argument 1 could not be converted to any of: sequence<sequence<ByteString>>, record<ByteString, ByteString>."
​​
stack: "mergeHeaders@http://localhost:3334/_nuxt/node_modules/ofetch/dist/shared/ofetch.4747642d.mjs?v=d0a42750:117:12\nresolveFetchOptions@http://localhost:3334/_nuxt/node_modules/ofetch/dist/shared/ofetch.4747642d.mjs?v=d0a42750:93:31\n$fetchRaw2@http://localhost:3334/_nuxt/node_modules/ofetch/dist/shared/ofetch.4747642d.mjs?v=d0a42750:194:16\n$fetch2@http://localhost:3334/_nuxt/node_modules/ofetch/dist/shared/ofetch.4747642d.mjs?v=d0a42750:304:21\ncall@http://localhost:3334/_nuxt/node_modules/@trandaison/nuxt-3-auth/dist/runtime/services/HttpService.mjs?v=d0a42750:59:28\nlogin@http://localhost:3334/_nuxt/node_modules/@trandaison/nuxt-3-auth/dist/runtime/services/Auth.mjs?v=d0a42750:60:46\nlogin@http://localhost:3334/_nuxt/node_modules/@trandaison/nuxt-3-auth/dist/runtime/composables/useLogin.mjs?v=d0a42750:47:18\nauthenticate@http://localhost:3334/_nuxt/pages/login.vue:66:28\n_sfc_render/_cache[3]<@http://localhost:3334/_nuxt/pages/login.vue:137:79\nwithModifiers/cache[cacheKey]@http://localhost:333…"
​​
<prototype>: TypeError.prototype { stack: "", … }
​
_value: TypeError: Headers constructor: Argument 1 could not be converted to any of: sequence<sequence<ByteString>>, record<ByteString, ByteString>.
​
dep: Object { computed: undefined, version: 1, sc: 0, … }
​
<prototype>: Object { … }
login.vue:132:12

```


So if the bug could not fixed in some time, maybe you can put the version 0.1.1 back to npm for a downgrade until the bug is fixed.
thx

**Comment by Stiropor** [comment]
Yeah, when using login from useLogin (as I did before, I have custom form but used the `Authenticator` just to test) I only get `An error has occured`. Then I tried $auth.login and I get these headers error in console. I didn't set any headers as I don't need any extra ones.

Yeah, 0.1.1 should be put back until this is solved.

I know this comment doesn't help much but I feel happy it's not only on our side 😃

**Comment by datafieber** [comment]
It is strange, if i clone the repo of v0.1.4 and update the environment to my local backend with the auth property from my  nuxt.config.ts, run the playground app it works.

**Comment by trandaison** [comment]
This might related to the latest release of ofetch https://github.com/unjs/ofetch/releases/tag/v1.4.0
> In the interceptors, the ctx.options.headers is now always normalized to [Headers](https://developer.mozilla.org/en-US/docs/Web/API/Headers).
> This is not a breaking change as ofetch >=1.4 will narrow 3 possibilities of HeadersInit to be always normalized Headers.
If you were previously not handling the Headers case and implicitly depending on input, you need to update the logic.

I will test with the latest ofetch and fix the issue.

> maybe you can put the version 0.1.1 back to npm for a downgrade

Unfortunately, I can't do that. npm does not allow to publish previous versions.

Can you try to downgrade the version of nuxt and ofetch to see if it works?

**Comment by datafieber** [comment]
I found some fix, if i add an empty headers object to all endpoints, it works.
 login: { url: "/auth/login", method: "post", headers: {} },
      logout: { url: "/auth/logout", method: "post", headers: {} },
      refresh: { url: "/auth/refresh", method: "post", headers: {} },
      user: { url: "/users/profile", method: "get", headers: {} },

**Comment by Stiropor** [comment]
> I found some fix, if i add an empty headers object to all endpoints, it works. login: { url: "/auth/login", method: "post", headers: {} }, logout: { url: "/auth/logout", method: "post", headers: {} }, refresh: { url: "/auth/refresh", method: "post", headers: {} }, user: { url: "/users/profile", method: "get", headers: {} },

@datafieber Sir, you saved the day 😃

**Comment by trandaison** [comment]
Hi @Stiropor @datafieber 

This should be fixed in [v0.1.5](https://github.com/trandaison/nuxt-3-auth/releases/tag/v0.1.5). Please update to v0.1.15 and let me know if there is any other errors.
